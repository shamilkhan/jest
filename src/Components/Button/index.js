import React from 'react';
import styled from 'styled-components/macro';

const Button = styled.button`
    padding: 5px 10px;
    border: 1px solid #000;
    outline: 0;
    background-color: transparent;
    border-radius: 4px;
    cursor: pointer;
`

function ButtonComponent({children, text}) {
    return (
        <Button>
            {children ? children : text ? text: ""}
        </Button>   
    )
}

export default ButtonComponent;