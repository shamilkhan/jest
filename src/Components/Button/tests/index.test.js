import React from 'react';
import Button from '../index';
import renderer from 'react-test-renderer';

it('renders <Button /> correctlys', () => {
  const tree = renderer
    .create(<Button text="test"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// it("<Button> component shows the same with props.text and props.children", () => {
//   const btn1 = renderer.create(<Button text="test"/>).toJSON();
//   const btn2 = renderer.create(<Button>test</Button>).toJSON();
//   expect(btn1).toMatchSnapshot();
//   expect(btn2).toMatchSnapshot();
//   expect(btn1).toEqual(btn2);
// });

