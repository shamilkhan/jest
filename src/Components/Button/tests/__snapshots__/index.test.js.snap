// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Button /> correctlys 1`] = `
<button
  className="Button-iopl9m-0 kmRaeJ"
>
  test
</button>
`;
